clients/admin-ui/src/features/system/VendorSelector.tsx

```diff
@@ -67,140 +66,137 @@ const CompassButton = ({
   const isInvalid = !!(meta.touched && meta.error);
   const field = { ...initialField, value: initialField.value ?? "" };
   const { touched, values, setTouched, setFieldValue, validateForm } =
     useFormikContext<FormValues>();
+  const [isTypeahead, setIsTypeahead] = useState(true);
 
   const selected = options.find((o) => o.value === field.value) ?? {
     label: field.value,
     value: field.value,
     description: "",
   };
 
+  const filterFunction = (searchParam: string, option?: DictOption) =>
+    !!option?.label.toLowerCase().startsWith(searchParam.toLowerCase());
+
   const [searchParam, setSearchParam] = useState<string>("");
 
-  const suggestions = options.filter((opt) =>
-    opt.label.toLowerCase().startsWith(searchParam.toLowerCase()),
+  const suggestions = useMemo(
+    () => options.filter((o) => filterFunction(searchParam, o)),
+    [options, searchParam],
   );
 
-  const isTypeahead = !field.value && !values.vendor_id;
+  const optionsWithCustom = useMemo(() => {
+    let o = options;
+    if (isCreate && searchParam) {
+      o = [
+        ...options,
+        {
+          label: `${NEW_SYSTEM_PREFIX} "${searchParam}"...`,
+          value: searchParam,
+        },
+      ];
+    }
+    return o;
+  }, [isCreate, options, searchParam]);
   const hasVendorSuggestions = !!searchParam && suggestions.length > 0;
   const nameFieldLockedForGVL = lockedForGVL && !isCreate;
 
+  useEffect(() => {
+    setIsTypeahead(!field.value && !values.vendor_id);
+  }, [field.value, values.vendor_id, setIsTypeahead]);
+
   useEffect(() => {
     validateForm();
   }, [isTypeahead, validateForm]);
 
-  const handleClear = () => {
-    setValue("");
+  const handleClear = async () => {
     setSearchParam("");
     setFieldValue("vendor_id", undefined);
+    await setValue("");
+    setTouched({ ...touched, vendor_id: false, name: false });
     onVendorSelected(undefined);
   };
 
-  const handleTabPressed = () => {
-    if (!searchParam) {
-      return;
-    }
-    if (suggestions.length > 0 && searchParam !== suggestions[0].label) {
-      setSearchParam(suggestions[0].label);
-      setValue(suggestions[0].value);
-    }
-  };
-
-  const handleSelectChange = (
-    newValue: SingleValue<Option>,
-    actionMeta: ActionMeta<Option>,
-  ) => {
-    if (actionMeta.action === "clear") {
-      handleClear();
-      return;
-    }
-    setValue(newValue ? newValue.label : "");
-    setTouched(
-      { ...touched, vendor_id: true, name: true },
-      // do not validate if a new option was created; this prevents
-      // incorrectly showing a "required field" error while a value is in
-      // the field
-      actionMeta.action !== "create-option",
-    );
+  const handleChange = async (newValue: DictOption) => {
     if (newValue) {
       const newVendorId = options.some((opt) => opt.value === newValue.value)
         ? newValue.value
         : undefined;
       setFieldValue("vendor_id", newVendorId);
+      await setValue(
```

Comment:
Didn't know you could explicitly await a state call, this is a much nicer solution to that error message bug.